app/src/main/java/org/gnucash/android/model/Transaction.java

```diff
@@ -268,7 +268,11 @@ public Money getBalance(String accountUID){
     public Money getImbalance(){
         Money imbalance = Money.createZeroInstance(mCurrencyCode);
         for (Split split : mSplitList) {
-            if (!split.getQuantity().getCommodity().getCurrencyCode().equals(mCurrencyCode)) {
+            Money quantity = split.getQuantity();
+            if (quantity.isAmountZero()) {
```

Comment:
I'd say `quantity` should never be zero (see [this comment](https://github.com/codinguser/gnucash-android/issues/506#issuecomment-231603278)). Am I missing something?